src/notes/service/NoteSender.swift

```diff
@@ -33,13 +33,21 @@ class DefaultNoteSender: NoteSender {
     }
     
     func sendNote(note: Note, viewController: UIViewController) {
-        // TODO - Figure out how to send the note as an email where
-        // the title is the subject and the note body is the email body
         
-        let alert = UIAlertController(title: "Coming soon", message: "Feature not yet implemented.", preferredStyle: .alert)
+        // set up activity view controller
+        let noteToShare = [note.body]
+        
+        let activityViewController = UIActivityViewController(activityItems: noteToShare as [Any], applicationActivities: nil)
+        
+        // this enables support for ipads
+        activityViewController.popoverPresentationController?.sourceView = viewController.view
+        
+        // set the subject line for emails
+        activityViewController.setValue(note.title, forKey: "Subject")
+        
+        // present the controller
+        viewController.present(activityViewController, animated: true, completion: nil)
```

Comment:
I added an extension in `extension/UIViewController.swift` that exposes a variable called `presentedVC` which is of type `UIViewController`. You may notice I always call `present` from that variable. The reason I do this is because a `UIViewController` at some point in execution might have an embedded / sub-viewcontroller that is presenting. If so, and you call present from the parent, and not from the currently presenting view controller, you will cause a hard crash. An example of this was the bug where clicking `+` while searching caused a hard crash. This was because the `UISearchController` was doing some presenting behind the scenes. So, the prompt to name the file would cause a hard crash because we tried to present from the parent, despite the fact that another child was already presenting. If no child is presenting, then the normal view controller is used.

Reply:
fixed